### PR TITLE
Support custom baseUrl.

### PR DESCRIPTION
**Pull request by tcobbs-bentley**

### CODE REVIEW

**Comment by tcobbs-bentley** [comment]
@aruniverse Assuming you are OK with this PR, can we get a new release with this functionality?

Note: our current mobile sample app has the following workaround for this missing functionality:

```ts
  const client = new ProjectsAccessClient();
  // ProjectsAccessClient doesn't have any public way to change its base URL. Hopefully that
  // will change in the future.
  (client as any)._baseUrl = baseUrl;
```

**Comment by tcobbs-bentley** [comment]
@aruniverse I think the test failures are due to the CI environment not containing a token, but I'm not sure about that.

**Comment by davidhjones** [comment]
@tcobbs-bentley @aruniverse any reason why we're updating this client? The Projects API is being deprecated (https://developer.bentley.com/apis/projects/#thisapiisbeingdeprecated), and all new development should be moved to the iTwins API. We have a client for calling the iTwins API: https://github.com/itwin/itwins-client.

**Comment by davidhjones** [comment]
> @aruniverse Assuming you are OK with this PR, can we get a new release with this functionality?
> 
> Note: our current mobile sample app has the following workaround for this missing functionality:
> 
> ```ts
>   const client = new ProjectsAccessClient();
>   // ProjectsAccessClient doesn't have any public way to change its base URL. Hopefully that
>   // will change in the future.
>   (client as any)._baseUrl = baseUrl;
> ```

In your code, what do you set the value of baseUrl to?

**Comment by aruniverse** [comment]
> @tcobbs-bentley @aruniverse any reason why we're updating this client? The Projects API is being deprecated ([developer.bentley.com/apis/projects/#thisapiisbeingdeprecated](https://developer.bentley.com/apis/projects/#thisapiisbeingdeprecated)), and all new development should be moved to the iTwins API. We have a client for calling the iTwins API: [itwin/itwins-client](https://github.com/itwin/itwins-client).

I am also in favor of switching to itwin api/client instead

File: src/ProjectsClient.ts [hunk 1]
```diff
@@ -16,12 +16,18 @@ import type { Project, ProjectsAccess, ProjectsLinks, ProjectsQueryArg, Projects
...
 
-  public constructor() {
-    const urlPrefix = process.env.IMJS_URL_PREFIX;
-    if (urlPrefix) {
-      const baseUrl = new URL(this._baseUrl);
-      baseUrl.hostname = urlPrefix + baseUrl.hostname;
-      this._baseUrl = baseUrl.href;
+  public constructor(baseUrl: string | undefined = undefined) {
+    if (baseUrl !== undefined) {
+      // Mobile apps do not have the ability to put things into process.env at run-time, so relying
```
[the remark this answers is not in the document]
Comment (tcobbs-bentley) [reply src/ProjectsClient.ts:21]: @aruniverse It looks like that will work. It's super ugly, and if this package weren't deprecated I'd say that this PR is worthy anyway. If this PR isn't going to be accepted (and a new release made), then I feel that this repository should be archived to indicate that no further changes will be accepted.

@davidhjones Regarding us updating to `@itwin/itwins-client`, I submitted a PR to that project to incorporate next paging support from the _links portion of the replies. It's possible to make use of that client without that paging support, but would require a fair bit of work on our part (and make the code uglier, IMO).

**Comment by tcobbs-bentley** [comment]
> @tcobbs-bentley In your code, what do you set the value of baseUrl to?

Here is the snippet again with one more line of code:

```ts
  const baseUrl = `https://${window.itmSampleParams.apiPrefix}api.bentley.com/projects/`;
  const client = new ProjectsAccessClient();
  // ProjectsAccessClient doesn't have any public way to change its base URL. Hopefully that
  // will change in the future.
  (client as any)._baseUrl = baseUrl;
```

When running in QA mode, it ends up being `https://qa-api.bentley.com/projects/`. When in prod mode, it's `https://api.bentley.com/projects/`.

**Pull request closed without merging**